api/src/services/authorization.ts

```diff
@@ -554,52 +546,146 @@ export class AuthorizationService {
 						_nnull: true,
 					},
 				});
+
+				hasValidationRules = true;
 			}
 		}
 
-		if (hasFieldValidationRules) {
-			if (permission.validation && Object.keys(permission.validation).length > 0) {
-				permission.validation = { _and: [permission.validation, ...fieldValidationRules] };
-			} else {
-				permission.validation = { _and: fieldValidationRules };
+		const { permissions, requiredPermissionData, containDynamicData } = parsePermissions([permission]);
+
+		return { permission: permissions[0], requiredPermissionData, containDynamicData, hasValidationRules };
+	}
+
+	/**
+	 * Checks if the provided payload matches the configured permissions, and adds the presets to the payload.
+	 */
+	async validatePayload(
+		action: PermissionsAction,
+		collection: string,
+		data: Partial<Item>,
+		pk?: PrimaryKey | PrimaryKey[]
+	): Promise<Array<PayloadChunk> | PayloadChunk> {
+		const payload = cloneDeep(data);
+
+		const { permission, requiredPermissionData, containDynamicData, hasValidationRules } = this.getActionPermissions(
+			action,
+			collection,
+			data
+		);
+
+		const filterContext: Partial<ParseFilterContext> =
+			containDynamicData && this.accountability
+				? await getFilterContext(this.schema, this.accountability, requiredPermissionData)
+				: {};
+
+		const payloadFields = Object.keys(payload);
+		const itemContextFields = requiredPermissionData.$CURRENT_ITEM.filter(
+			(field: string) => !payloadFields.includes(field)
+		);
+
+		const pkField = this.schema.collections[collection].primary;
+		const includesPK = pk && itemContextFields.includes(pkField);
+		let itemContexts: Array<Partial<Item>> = [
+			{
+				[pkField]: '+',
+			},
+		];
+
+		if (pk) {
+			if (!includesPK) {
+				itemContextFields.push(pkField);
 			}
+
+			itemContexts = await this.getItemsContext(action, collection, pk, itemContextFields);
+
+			// Remove pk field so we don't check it during hashing later on
+			itemContextFields.splice(itemContextFields.indexOf(pkField), 1);
 		}
 
-		const validationErrors: FailedValidationException[] = [];
+		const chunks: Record<string, PayloadChunk> = {};
+		itemContexts.forEach((itemContext) => {
+			itemContext = merge({}, itemContext, payload);
```

Comment:
Hey, just looking over this as I was looking to implement something similar, and I think a potential issue you could run into here is the detailed syntax for related items, i.e.
```js
{ 
   create: [],
   update: [],
   delete: [],
}
```
showing up in the payload, and therefore not properly merging with the itemContext. Especially if there are some nested updates that need to be validated. I guess you'd need to merge updated items, add created items, and remove deleted items from the itemContext... recursively?

Reply:
Similar to this issue: https://github.com/directus/directus/pull/17842